### PR TITLE
Recognize ubuntu 20.10

### PR DESCRIPTION
**Pull request by justinsb**
Teach nodeup about ubuntu 20.10, including the unusual
/etc/resolv.conf configuration.

### CODE REVIEW

**Comment by justinsb** [comment]
I'm actually not sure if we want to recognize / support non-LTS ubuntu versions.  It's pretty low-effort now that we're installing docker / containerd from tar.gz packages.

**Comment by hakman** [comment]
> I'm actually not sure if we want to recognize / support non-LTS ubuntu versions. It's pretty low-effort now that we're installing docker / containerd from tar.gz packages.

As long as we agree that we will treat these non-LTS as LATEST and always replace it when a new one is released, should be fine.

File: util/pkg/distributions/identify_test.go [hunk 1]
```diff
@@ -99,6 +99,11 @@ func TestFindDistribution(t *testing.T) {
 			err:      nil,
 			expected: DistributionUbuntu2004,
 		},
+		{
+			rootfs:   "ubuntu2010",
+			err:      nil,
+			expected: DistributionUbuntu2010,
+		},
```
Comment (hakman) [inline util/pkg/distributions/identify_test.go:106]: The tests expect the `/etc/os-release` from an Ubuntu 20.10 to be copied to the `tests` dir.

File: nodeup/pkg/model/miscutils.go [hunk 2]
```diff
@@ -48,8 +48,15 @@ func (b *MiscUtilsBuilder) Build(c *fi.ModelBuilderContext) error {
 			packages = append(packages, "curl")
 			packages = append(packages, "wget")
 			packages = append(packages, "perl")
-			packages = append(packages, "python-apt")
 			packages = append(packages, "apt-transport-https")
+
+			// TODO: Do we really need python-apt?
```
Comment (johngmyers) [inline nodeup/pkg/model/miscutils.go:53]: I doubt we need any of these, which is why I'm phasing them out for k8s 1.20+

**Comment by justinsb** [comment]
> As long as we agree that we will treat these non-LTS as LATEST and always replace it when a new one is released, should be fine.

I'm going to put this on the agenda for today.  OTOH I agree that otherwise it could become a lot of work, OTOH we don't really want a kops upgrade to break people's working configurations.

What motivated this was that the latest ubuntu image on GCP is not LTS (the LTS is also available, it just wasn't the first that caught my eye).  So it was a little annoying to have the cluster fail to start there.

**Comment by hakman** [comment]
Cool! :)
/lgtm

**Comment by MoShitrit** [comment]
@hakman Does this mean we can start updating the ami with 20.10 versions on a more frequent basis from now on?

**Comment by olemarkus** [comment]
@MoShitrit If you refer to the channels, they should remain with the LTS versions.

**Comment by MoShitrit** [comment]
Thanks for clarifying @olemarkus !

**Pull request merged**